core/txpool/txpool.go

```diff
@@ -632,14 +635,32 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 		return core.ErrTipAboveFeeCap
 	}
 	// Make sure the transaction is signed properly.
-	from, err := types.Sender(pool.signer, tx)
-	if err != nil {
+	if _, err := types.Sender(pool.signer, tx); err != nil {
 		return ErrInvalidSender
 	}
 	// Drop non-local transactions under our own minimal accepted gas price or tip
 	if !local && tx.GasTipCapIntCmp(pool.gasPrice) < 0 {
 		return ErrUnderpriced
 	}
+	// Ensure the transaction has more gas than the basic tx fee.
+	intrGas, err := core.IntrinsicGas(tx.Data(), tx.AccessList(), tx.To() == nil, true, pool.istanbul.Load(), pool.shanghai.Load())
+	if err != nil {
+		return err
+	}
+	if tx.Gas() < intrGas {
+		return core.ErrIntrinsicGas
+	}
+	return nil
+}
+
+// validateTx checks whether a transaction is valid according to the consensus
+// rules and adheres to some heuristic limits of the local node (price and size).
+func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
+	if err := pool.validateTxBasics(tx, local); err != nil {
+		return err
+	}
```

Comment:
honestly I feel this check is unnecessary. I think the check in txpool doesn't need to be very strict.

Like the init-code-size check raised in shanghai, it's totally possible that the "large-code" transactions are accepted before the fork and stay in pool for long time because of low gas price.

So another question is: if these transactions are accepted into pool, how can they get evicted? Feel like they will stay until get evicted because of low price.